### PR TITLE
Feature/modal-validation-and-UI-fixes-updates

### PR DESCRIPTION
**Pull request by ShubhamThakre**
## Checklist

- **Modal validations added;**

    - Groups Tab -> Create New Groups -> validations added to modal fields. 
     Screenshot 2022-03-17 at 7 07 13 PM
     
    - Ingestions -> Secrets Tabs -> Create New Secrets -> validations added to modal fields.
     Screenshot 2022-03-17 at 7 06 58 PM
     
    - Domains -> New Domains -> validations added to modal fields.   
      Screenshot 2022-03-17 at 7 06 38 PM
 
-  **Glossary Term** -> Related Terms tab -> Added EmbeddedListSearch component with fixedQuery Filter
 Screenshot 2022-03-17 at 7 10 22 PM

-  **UI fixes;**

 - Search bar Autocorrect Issue resolved.
 - Groups -> Sidebard member avatar is not matching with Members tabs avatar. Issue resolved.
Screenshot 2022-03-17 at 7 12 28 PM

@jjoyce0510

### CODE REVIEW

File: datahub-web-react/src/app/search/SearchBar.tsx [hunk 1]
```diff
@@ -278,13 +278,15 @@ export const SearchBar = ({
...
-                        e.stopPropagation();
+                    onPressEnter={() => {
+                        // e.stopPropagation();
                         onSearch(filterSearchQuery(searchQuery || ''));
                     }}
                     style={inputStyle}
                     value={searchQuery}
-                    onChange={(e) => setSearchQuery(e.target.value)}
+                    onChange={(e) => {
+                        setSearchQuery(e.target.value);
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/search/SearchBar.tsx:288]: why is this change made?

File: datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnerModal.tsx [hunk 2]
```diff
@@ -195,6 +195,10 @@ export const AddOwnerModal = ({ urn, type, visible, hideOwnerType, defaultOwnerT
     //    querySelectorToExecuteClick: selectedActor && '#addOwnerButton',
     // });
 
+    // console.log(makeid(5));
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnerModal.tsx:198]: We can remove this :)

File: datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnerModal.tsx [hunk 3]
```diff
@@ -228,7 +232,7 @@ export const AddOwnerModal = ({ urn, type, visible, hideOwnerType, defaultOwnerT
                         tagRender={(tagProps) => <Tag>{tagProps.value}</Tag>}
                     >
                         {combinedSearchResults?.map((result) => (
-                            <Select.Option value={result.entity.urn}>{renderSearchResult(result)}</Select.Option>
+                            <Select.Option value={result.entity.urn}> {renderSearchResult(result)}</Select.Option>
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnerModal.tsx:235]: Do we need a key here also?

File: datahub-web-react/src/app/ingest/secret/SecretBuilderModal.tsx [hunk 4]
```diff
@@ -53,48 +57,82 @@ export const SecretBuilderModal = ({ initialState, visible, onSubmit, onCancel }
...
+                <Form.Item label={<Typography.Text strong>Description</Typography.Text>}>
                     <Typography.Paragraph>
                         An optional description to help keep track of your secret.
                     </Typography.Paragraph>
-                    <Input
-                        placeholder="The value of your secret"
-                        value={secretBuilderState.description}
-                        onChange={(event) => setDescription(event.target.value)}
-                    />
+                    <Form.Item name="description" rules={[{ whitespace: true }, { min: 1, max: 500 }]} hasFeedback>
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/ingest/secret/SecretBuilderModal.tsx:129]: Why do we need `whitespace: true` here?

**Review by jjoyce0510 — COMMENTED** [review]
Overall looks good - just minor comments.

File: datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnerModal.tsx (hunk 2 above)
Comment (ShubhamThakre) [reply datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnerModal.tsx:198]: Sure.

File: datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnerModal.tsx (hunk 3 above)
Comment (ShubhamThakre) [reply datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnerModal.tsx:235]: Yes, we do required the key property here. We have added the separate function in which will help to generate the unique key for the datasets of combinedSearchResults.

File: datahub-web-react/src/app/ingest/secret/SecretBuilderModal.tsx (hunk 4 above)
Comment (ShubhamThakre) [reply datahub-web-react/src/app/ingest/secret/SecretBuilderModal.tsx:129]: User should not able to add only space as text, either it should be blank or some text, not just white space

File: datahub-web-react/src/app/search/SearchBar.tsx (hunk 1 above)
Comment (ShubhamThakre) [reply datahub-web-react/src/app/search/SearchBar.tsx:288]: We were testing something on it. It's not changed to previous one.

File: datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnerModal.tsx [hunk 5]
```diff
@@ -213,41 +215,47 @@ export const AddOwnerModal = ({ urn, type, visible, hideOwnerType, defaultOwnerT
...
+                            mode="multiple"
+                            ref={inputEl}
+                            placeholder="Search for users or groups..."
+                            onSelect={(actorUrn: any) => onSelectActor(actorUrn)}
+                            onDeselect={(actorUrn: any) => onDeselectActor(actorUrn)}
+                            onSearch={handleActorSearch}
+                            tagRender={(tagProps) => <Tag>{tagProps.value}</Tag>}
+                        >
+                            {combinedSearchResultsUpdated?.map((result) => (
+                                <Select.Option key={result.uniqueId} value={result.entity.urn}>
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/AddOwnerModal.tsx:234]: Key can just be... result.entity.urn

File: datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/SidebarOwnerSection.tsx [hunk 6]
```diff
@@ -18,7 +18,12 @@ export const SidebarOwnerSection = ({ properties }: { properties?: any }) => {
             <SidebarHeader title="Owners" />
             <div>
                 {entityData?.ownership?.owners?.map((owner) => (
-                    <ExpandedOwner entityUrn={urn} owner={owner} refetch={refetch} />
+                    <ExpandedOwner
+                        key={owner.owner.editableProperties?.email || owner.owner.properties?.displayName}
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/SidebarOwnerSection.tsx:22]: let's use "urn" here

**Review by jjoyce0510 — APPROVED** [review]

**Pull request merged**